Objects/funcobject.c

```diff
@@ -3,12 +3,58 @@
 
 #include "Python.h"
 #include "pycore_ceval.h"         // _PyEval_BuiltinsFromGlobals()
+#include "pycore_function.h"      // FUNC_MAX_WATCHERS
 #include "pycore_object.h"        // _PyObject_GC_UNTRACK()
 #include "pycore_pyerrors.h"      // _PyErr_Occurred()
 #include "structmember.h"         // PyMemberDef
 
 static uint32_t next_func_version = 1;
 
+static void
+handle_func_event(PyFunction_WatchEvent event, PyFunctionObject *func, PyObject *new_value)
+{
+    PyInterpreterState *interp = _PyInterpreterState_GET();
+    for (int i = 0; i < FUNC_MAX_WATCHERS; i++) {
```

Comment:
If we add a bit vector we can skip the iteration when there are no watchers.
Maybe put that test in an inline function and call this function if there are watchers?